win/WIN_ButtonGroup.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <vector>
+#include <memory>
+
+namespace win
+{
+
+	class ToggleButton;
+	
+	class ButtonGroup
+	{
+	public:
+		ButtonGroup();
+		~ButtonGroup() = default;
+		ButtonGroup(const ButtonGroup& that) = default;
+		ButtonGroup(ButtonGroup&& that) = default;
+		ButtonGroup& operator=(const ButtonGroup& that) = default;
+		ButtonGroup& operator=(ButtonGroup&& that) = default;
+
+		void addButtonChild(std::shared_ptr<ToggleButton> const& child);
+		std::vector<std::shared_ptr<ToggleButton>> getButtonChildren() const { return buttonChildren_; }
+		void turnOffOtherChildren();
+		void setSelectedChild(ToggleButton* child);
+		
+
+	private:
+		std::vector<std::shared_ptr<ToggleButton>> buttonChildren_;
```

Comment:
Possibly should be `std::weak_ptr<ToggleButton>` though in a static case such as ours where no buttons are added / removed we might get away with it. Check by putting print statements in the ToggleButton destructors. If they don't print when the program is closed, change to the weak version.

Reply:
Done - changed to weak_ptr as they were not being destroyed